Tutorials/CNTK_209A_Evaluation_of_Pretrained_Super-resolution_Models.ipynb

```diff
@@ -0,0 +1,634 @@
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import urllib\n",
+    "try:\n",
+    "    from urllib.request import urlretrieve, urlopen\n",
+    "except ImportError: \n",
+    "    from urllib import urlretrieve, urlopen\n",
+    "\n",
+    "link = \"https://www2.eecs.berkeley.edu/Research/Projects/CS/vision/grouping/segbench/BSDS300/html/images/plain/normal/color/253027.jpg\"\n",
+    "\n",
+    "urlretrieve(link, os.path.join(example_folder, \"example_1.jpg\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "We will also save a copy of bicubic interpolation effect for every image we are testing on, just for reference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "save_folder = os.path.join(results_folder, \"bicubic\")\n",
+    "\n",
+    "#upscale by bicubic and save for reference\n",
+    "for entry in os.listdir(example_folder):\n",
+    "    filename = os.path.join(example_folder, entry)\n",
+    "        \n",
+    "    if not os.path.exists(save_folder):\n",
+    "        os.makedirs(save_folder)\n",
+    "    \n",
+    "    img = Image.open(filename)\n",
+    "    out = img.resize((2 * img.width, 2 * img.height), Image.BICUBIC)\n",
+    "    out.save(os.path.join(save_folder, entry))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "<p>Now we are finally ready to evaluate our models. The code below combines all possible filters for all images. Parameters in the calls of <code>evaluate</code> are set according to the models in question. For example, models on indices 0 and 1 are learning only the residual image, so we set <code>residual_model = True</code>.</p>\n",
+    "<p>Also, all upscaling is done in the first element of the filter. Therefore, only <b>VDSR</b> and <b>DRNN</b> can be the second element of the filter and their preprocess part must be skipped in that case. We process the images and save them in appropriate folders. Folder names reflect which filter was used. See code below for additional comments.</p>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Now creating: Data/BerkeleySegmentationDataset/example_results/VDSR_results/example_1.jpg\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/anaconda/envs/py35/lib/python3.5/site-packages/cntk/core.py:74: RuntimeWarning: data is not C contiguous; rearrange your data/computation to avoid costly data conversions\n",
```

Comment:
This runtime warning appears, not sure how to avoid it. It does not affect the performance or running of the notebook.